app/routes.py

```diff
@@ -1 +1,272 @@
-from flask import Blueprint
\ No newline at end of file
+from os import abort
+from datetime import datetime
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request, abort, make_response
+import sqlalchemy
+import os
+import requests
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_one_task():
+    if not request.is_json:
+        return {"msg": "Missing JSON request body"}, 400
+
+    request_body = request.get_json()
+    try:
+        title = request_body["title"]
+        description = request_body["description"]
+        completed_at = request_body["completed_at"]
+
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+
+    new_task = Task(
+        title=title,
+        description=description,
+        completed_at=completed_at
+        if request_body["completed_at"] is not None else None
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    rsp = {"task": new_task.get_dict()}
+
+    return jsonify(rsp), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    tasks = Task.query.all()
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(sqlalchemy.asc(Task.title))
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(sqlalchemy.desc(Task.title))
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.get_dict())
+
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_task(task_id)
+    rsp = {"task" : task.get_dict()}
+
+    return jsonify(rsp), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = validate_task(task_id)
+
+    if not request.is_json:
+        return {"msg" : "Missing JSON request body"}, 400
+
+    request_body = request.get_json()
+    try:
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+    except KeyError:
+        return {
+            "msg": "Update failed due to missing data. Title, Description are required!"
+        }, 400
```

Comment:
This try and except is great for if something fails 🏁 Nice safety net just in case